var/spack/repos/builtin/packages/rocm-tensile/package.py

```diff
@@ -62,8 +64,20 @@ class RocmTensile(CMakePackage):
     def setup_build_environment(self, env):
         env.set('CXX', self.spec['hip'].hipcc)
 
-    def cmake_args(self):
+    def get_gpulist_for_tensile_support(self):
         arch = self.spec.variants['tensile_architecture'].value
+        if arch[0] == 'all':
+            if self.spec.satisfies('@:4.0.0'):
+                arch_value = self.tensile_architecture[1:2] + 'gfx906,gfx908'
+            elif self.spec.satisfies('@4.1.0:4.2.0'):
+                arch_value = self.tensile_architecture[1:4]
+            elif self.spec.satisfies('@4.3.0:'):
+                arch_value = self.tensile_architecture[1:]
+            return arch_value
```

Comment:
With this PR I don't think it would be possible to specify: `@4.0.0 tensile_architecture=gfx906`, so would it be possible to do

```
tensile_architecture = ('all', 'gfx000', 'gfx900', 'gfx906', 'gfx908', 'gfx906:xnack-', 'gfx908:xnack-',
                             'gfx90a:xnack-', 'gfx1010', 'gfx1011', 'gfx1012', 'gfx1030')
conflicts('tensile_architecture=gfx906', when='@4.0.1:')
...
```

?

That would complicate `get_gpulist_for_tensile_support` although IMO with some minor redundancy it's not too bad:

```
def all_archs_for_version(self):
  spec = self.spec
  # Versions after 4.0.0 refer to gfx906 as gfx906:xnack-
  arch_refs = ('gfx000', 'gfx900', 'gfx906:xnack-', 'gfx908:xnack-', 'gfx90a:xnack-', 'gfx1010', 'gfx1011', 'gfx1012', 'gfx1030')
  if spec.satisfies('@:4.0.0'):
    return (gfx000, gfx900, gfx906, gfx 908)
  elf spec.satisfies('@4.1.0:4.2.0'):
    return arch_refs[:4]
  else:
    return arch_refs
```

Reply:
yeah..i think for so..the best option for 4.0.0 is the default option of all so that it builds gfx906;gfx908. I did feel it would be much simpler to just duplicate the correct list for each release version rather than making a reader try to follow the list editing logic ..but that would mean some duplication of gpu list . i will add the conflicts () for now .

Reply:
I think it still isn't possible to say `@4.0.0 tensile_architecture=gfx906` with the latest changes, which is odd to me because `all` includes `gfx906` for `@:4.0.0`. Would you be able to edit the variant declaration to include `gfx906` (in addition to `gfx906:xnack-`)?